src/Autofac.Diagnostics.DotGraph/DotStringExtensions.cs

```diff
@@ -0,0 +1,246 @@
+        /// </param>
+        /// <param name="exceptionType">
+        /// The full exception type name to display.
+        /// </param>
+        /// <param name="exceptionMessage">
+        /// The message from the exception.
+        /// </param>
+        /// <returns>
+        /// The <paramref name="stringBuilder" /> for continued writing.
+        /// </returns>
+        public static StringBuilder AppendTableErrorRow(this StringBuilder stringBuilder, string exceptionType, string exceptionMessage)
+        {
+            stringBuilder.Append("<tr><td><font point-size=\"10\"><b>");
+            stringBuilder.Append(exceptionType.Encode());
+            stringBuilder.Append("</b>:<br/>\n");
+            stringBuilder.Append(exceptionMessage.Wrap().Encode());
+            stringBuilder.Append("</font></td></tr>");
+            stringBuilder.AppendLine();
+            return stringBuilder;
+        }
+
+        /// <summary>
+        /// Writes a DOT graph connection between two named nodes.
+        /// </summary>
+        /// <param name="stringBuilder">
+        /// The <see cref="StringBuilder" /> to which the node should be written.
+        /// </param>
+        /// <param name="fromId">
+        /// The ID of the node where the connection originates.
+        /// </param>
+        /// <param name="toId">
+        /// The ID of the node where the connection ends.
+        /// </param>
+        /// <param name="label">
+        /// The label on the connecting line.
+        /// </param>
+        /// <param name="bold">
+        /// <see langword="true"/> if the text in the cell should be bold; <see langword="false"/> if not.
+        /// </param>
+        /// <returns>
+        /// The <paramref name="stringBuilder" /> for continued writing.
+        /// </returns>
+        public static StringBuilder ConnectNodes(this StringBuilder stringBuilder, string fromId, string toId, string label, bool bold)
+        {
+            stringBuilder.AppendFormat(CultureInfo.CurrentCulture, "{0} -> {1} [label=<{2}>", fromId, toId, label.Encode());
+            if (bold)
+            {
+                stringBuilder.Append(",penwidth=3,color=red");
+            }
+
+            stringBuilder.AppendLine("]");
+            return stringBuilder;
+        }
+
+        /// <summary>
+        /// HTML-encodes and converts newlines to line break tags in an input string.
+        /// </summary>
+        /// <param name="input">
+        /// The <see cref="string"/> to encode.
+        /// </param>
+        /// <returns>
+        /// The encoded version of <paramref name="input"/>.
+        /// </returns>
+        public static string Encode(this string input)
+        {
+            // Pretty stoked the StringComparison overload is only in one of our
+            // target frameworks. :(
+#if NETSTANDARD2_0
+                return HttpUtility.HtmlEncode(input).Replace(Environment.NewLine, "<br/>");
+#endif
```

Comment:
I went with `if !NETSTANDARD2_0` to avoid future maintenance problems. Like, if we drop netstandard 2.x and go straight to netcoreapp3 or something, technically this will all still compile correctly. No need to search for specific references.

Reply:
Ok, gotcha.